discord/message.py

```diff
@@ -2415,6 +2474,69 @@ def system_content(self) -> str:
         if self.type is MessageType.guild_incident_report_false_alarm:
             return f'{self.author.name} reported a false alarm in {self.guild}.'
 
+        if self.type is MessageType.call:
+            call_ended = self.call.ended_timestamp is not None  # type: ignore # call can't be None here
+            missed = self._state.user not in self.call.participants  # type: ignore # call can't be None here
+
+            if call_ended:
+                seconds = self.call.duration.total_seconds()  # type: ignore # call can't be None here
+
+                minutes_s = 60
+                hours_s = minutes_s * 60
+                days_s = hours_s * 24
+                # Discord uses approx. 1/12 of 365.25 days (avg. days per year)
+                months_s = days_s * 30.4375
+                years_s = months_s * 12
+
+                threshold_s = 45
+                threshold_m = 45
+                threshold_h = 21.5
+                threshold_d = 25.5
+                threshold_M = 10.5
+
+                if seconds < threshold_s:
+                    duration = "a few seconds"
+                elif seconds < (threshold_m * minutes_s):
+                    minutes = round(seconds / minutes_s)
+                    if minutes == 1:
+                        duration = "a minute"
+                    else:
+                        duration = f"{minutes} minutes"
+                elif seconds < (threshold_h * hours_s):
+                    hours = round(seconds / hours_s)
+                    if hours == 1:
+                        duration = "an hour"
+                    else:
+                        duration = f"{hours} hours"
+                elif seconds < (threshold_d * days_s):
+                    days = round(seconds / days_s)
+                    if days == 1:
+                        duration = "a day"
+                    else:
+                        duration = f"{days} days"
+                elif seconds < (threshold_M * months_s):
+                    months = round(seconds / months_s)
+                    if months == 1:
+                        duration = "a month"
+                    else:
+                        duration = f"{months} months"
+                else:
+                    years = round(seconds / years_s)
+                    if years == 1:
+                        duration = "a year"
+                    else:
+                        duration = f"{years} years"
```

Comment:
I forgot to mention this on Discord (sorry) but this should have been a util function instead of bloating up the size of this one.

Reply:
Done